task/git-clone-oci-ta/0.1/git-clone-oci-ta.yaml

```
@@ -261,7 +261,17 @@ spec:
 
         if [ "${PARAM_FETCH_TAGS}" = "true" ]; then
           echo "Fetching tags"
-          git fetch --tags
+          # Checking, if tags get fetched, when repo has submodules
+          if ! git fetch --tags; then
+            echo "Retrying fetch command..."
+            if git fetch --tags; then
+              echo "Fetching tags was successful"
+            else
+              EXIT_CODE="$?"
+              echo "fatal: Fetching tags was not successful, exit code: ${EXIT_CODE}"
+              exit "${EXIT_CODE}"
+            fi
```

IMO we can simplify this block to just this

```suggestion
            git fetch --tags
```

The success message doesn't seem necessary (and it's a bit weird that we would print it after a successful retry but not if the first attempt already succeeds). And for the error message, we can just let git print its own

---

Thank you for a great review, I really appreciate it :smiley: